packages/nextjs-mf/src/plugins/NextFederationPlugin.ts

```diff
@@ -38,187 +52,470 @@ const regexEqual = (x, y) => {
+    asyncChunkLoading: true,
+    name: options.name,
+    remotes: options.remotes as Record<string, string>,
+    baseURI: compiler.options.output.publicPath,
+    verbose: true,
+  }).apply(compiler);
+}
+
+/**
+ * Gets the appropriate ModuleFederationPlugin based on the environment.
+ * @param {boolean} isServer - A flag to indicate if the environment is server-side or not.
+ * @param {Compiler} compiler - The Webpack compiler instance.
+ * @returns {ModuleFederationPlugin | undefined} The ModuleFederationPlugin or undefined if not applicable.
+ */
+function getModuleFederationPluginConstructor(
+  isServer: boolean,
+  compiler: Compiler
+): ConstructableModuleFederationPlugin {
+  if (isServer) {
+    return require('@module-federation/node')
+      .NodeFederationPlugin as ConstructableModuleFederationPlugin;
+  } else {
+    return compiler.webpack.container
+      .ModuleFederationPlugin as unknown as ConstructableModuleFederationPlugin;
+  }
+}
+
+/**
+ * Set up default shared values based on the environment.
+ * @param isServer - Boolean indicating if the code is running on the server.
+ */
+const setDefaultShared = (isServer: boolean) => {
+  return isServer ? DEFAULT_SHARE_SCOPE : DEFAULT_SHARE_SCOPE_BROWSER;
+};
+
+/**
+ * Inject module hoisting system.
+ */
+function injectModuleHoistingSystem(isServer: boolean, compiler: Compiler) {
+  const defaultShared = setDefaultShared(isServer);
+  // Inject hoist dependency into the upper scope of the application
+  const injectHoistDependency = {
+    enforce: 'pre',
+    test: /_document/,
+    include: [compiler.context, /next[\\/]dist/],
+    loader: path.resolve(__dirname, '../loaders/inject-hoist'),
+  };
+
+  // Populate hoist dependency with shared modules
+  const populateHoistDependency = {
+    test: /internal-delegate-hoist/,
+    include: [/internal-delegate-hoist/, compiler.context, /next[\\/]dist/],
+    loader: path.resolve(__dirname, '../loaders/share-scope-hoist'),
+    options: {
+      shared: defaultShared,
+      name: !isServer && options.name,
+    },
+  };
+
+  compiler.options.module.rules.unshift(
     //@ts-ignore
-    if (!compiler.options.name) {
-      throw new Error('name is not defined in Compiler options');
-    }
+    injectHoistDependency,
+    populateHoistDependency
+  );
+
+  return {
+    applyAutomaticAsyncBoundary: () => {
+      // Implementation of applyAutomaticAsyncBoundary
+    },
+    applyRemoteDelegates: () => {
+      // Implementation of applyRemoteDelegates
+    },
+  };
+}
+
+/**
+ * Apply automatic async boundary.
+ */
+function applyAutomaticAsyncBoundary(compiler: Compiler) {
+  const allowedPaths = ['pages/', 'app/', 'src/pages/', 'src/app/'];
 
-    if (!this._options.filename) {
-      throw new Error('filename is not defined in NextFederation options');
+  const jsRules = compiler.options.module.rules.find((r) => {
+    //@ts-ignore
+    return r && r.oneOf;
+  });
+
+  //@ts-ignore
+  if (jsRules && 'oneOf' in jsRules) {
```

Comment:
I think we don't have to check of 'oneOf' here as the above line:349 is already do that.

Reply:
check it again i updated some things